File: app/src/main/java/com/example/inappcamerademo/ImageViewBaseAdapter.java
package com.example.inappcamerademo;

import android.app.Activity;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import com.example.inappcamerademo.model.ImageInfo;

import java.io.File;
import java.io.FileInputStream;
import java.util.List;

public class ImageViewBaseAdapter extends BaseAdapter {

    private List<String> imageFileList;
    private Context context;
    private Activity activity;

    public ImageViewBaseAdapter(Context context, Activity activity, List<String> imageFileNames) {
        this.context = context;
        this.activity = activity;
        this.imageFileList = imageFileNames;
    }

    @Override
    public int getCount() {
        return imageFileList.size();
    }

    @Override
    public Object getItem(int i) {
        return imageFileList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(final int i, View view, ViewGroup viewGroup) {
        if (view == null) {
            view = LayoutInflater.from(context).inflate(R.layout.captured_image_view, viewGroup, false);
        }
        try {
            final ImageView imageView = view.findViewById(R.id.image);
            MainDbHelper mainDbHelper = MainDbHelper.getInstance(context);

            final ImageInfo imageInfo = mainDbHelper.getImageInfoFromImageName(imageFileList.get(i));

            new Thread(new Runnable() {
                @Override
                public void run() {
                    Bitmap bitmap = null;
                    try {
                        BitmapFactory.Options ops = new BitmapFactory.Options();
                        ops.inSampleSize = 16;
                        bitmap = BitmapFactory.decodeStream(new FileInputStream(new File(imageInfo.getPath())), null, ops);
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                    final Bitmap finalBitmap = bitmap;
                    activity.runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            imageView.setImageBitmap(finalBitmap);
                        }
                    });
                }
            }).start();


        } catch (Exception e) {
            e.printStackTrace();
        }
        return view;
    }

}

File: app/src/main/java/com/example/inappcamerademo/ImageViewBaseAdapter.kt
package com.example.inappcamerademo

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import java.io.File
import java.io.FileInputStream

class ImageViewBaseAdapter(
    private val context: Context,
    private val activity: Activity,
    private val imageFileList: List<String>
) : BaseAdapter() {
    override fun getCount(): Int {
        return imageFileList.size
    }

    override fun getItem(i: Int): Any {
        return imageFileList[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        var view = view
        if (view == null) {
            view =
                LayoutInflater.from(context).inflate(R.layout.captured_image_view, viewGroup, false)
        }
        try {
            val imageView = view.findViewById<ImageView>(R.id.image)
            val mainDbHelper = MainDbHelper.getInstance(context)
            val imageInfo = mainDbHelper?.getImageInfoFromImageName(imageFileList[i])
            Thread {
                var bitmap: Bitmap? = null
                try {
                    val ops = BitmapFactory.Options()
                    ops.inSampleSize = 16
                    bitmap = BitmapFactory.decodeStream(
                        FileInputStream(File(imageInfo?.path)),
                        null,
                        ops
                    )
                } catch (e: Exception) {
                    e.printStackTrace()
                }
                val finalBitmap = bitmap
                activity.runOnUiThread { imageView.setImageBitmap(finalBitmap) }
            }.start()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return view
    }
}